include/crocoddyl/core/squashing/smooth-sat.hpp

```diff
@@ -0,0 +1,85 @@
+#ifndef CROCODDYL_CORE_SQUASHING_SMOOTH_SAT_HPP_
+#define CROCODDYL_CORE_SQUASHING_SMOOTH_SAT_HPP_
+
+#include "crocoddyl/core/squashing-base.hpp"
+
+namespace crocoddyl {
+
+template <typename _Scalar>
+class SquashingModelSmoothSatTpl : public SquashingModelAbstractTpl<_Scalar> {
+ public:
+  typedef _Scalar Scalar;
+  typedef MathBaseTpl<Scalar> MathBase;
+  typedef SquashingModelAbstractTpl<Scalar> Base;
+  typedef SquashingDataAbstractTpl<Scalar> SquashingDataAbstract;
+  typedef typename MathBase::VectorXs VectorXs;
+
+  explicit SquashingModelSmoothSatTpl(const Eigen::Ref<const VectorXs>& out_lb,
+                                      const Eigen::Ref<const VectorXs>& out_ub, const std::size_t& ns)
+      : Base(ns) {
+    out_lb_ = out_lb;
+    out_ub_ = out_ub;
+
+    in_lb_ = out_lb_;
+    in_ub_ = out_ub_;
+
+    smooth_ = 0.1;
+
+    d_ = (out_ub_ - out_lb_) * smooth_;
+    a_ = d_.array() * d_.array();
+  }
+
+  ~SquashingModelSmoothSatTpl();
+
+  void calc(const boost::shared_ptr<SquashingDataAbstract>& data, const Eigen::Ref<const VectorXs>& u) {
+    // Squashing function used: "Smooth abs":
+    // s(u) = 0.5*(lb + ub + sqrt(smooth + (u - lb)^2) - sqrt(smooth + (u - ub)^2))
+    data->s =
+        0.5 * (Eigen::sqrt(Eigen::pow((u - out_lb_).array(), 2) + a_.array()) -
+               Eigen::sqrt(Eigen::pow((u - out_ub_).array(), 2) + a_.array()) + out_lb_.array() + out_ub_.array());
+  }
+
+  void calcDiff(const boost::shared_ptr<SquashingDataAbstract>& data, const Eigen::Ref<const VectorXs>& u,
+                const bool& recalc = true) {
+    if (recalc) {
```

Comment:
Remove the recalc if-loop, i.e. these lines of code:
```cpp
if (recalc) {
  calc(data, u);
}
```

Note that we have changed all the `recalc` option.
You know will need to run manually `calc` for computing `calcDiff`.